app/src/main/java/com/babylon/wallet/android/presentation/onboarding/restore/mnemonics/RestoreMnemonicsScreen.kt

```diff
@@ -73,16 +73,41 @@ fun RestoreMnemonicsScreen(
     RestoreMnemonicsContent(
         state = state,
         onBackClick = viewModel::onBackClick,
-        onSkipClicked = viewModel::onSkipClick,
-        onSubmitClick = {
-            if (!state.isShowingEntities) {
+        onSkipSeedPhraseClick = {
+            if (state.isLastRecoverableFactorSource) {
```

Comment:
> After we skipped seed phrase entry for the first time, I assume new main BDFS was created. Why we are asking for biometrics again is we open recovery screen and skip again? I don't think we should

@jakub-rdx So when you skip main seed phrase entry and there are more factor sources to restore we do not create BDFS yet. I defer creating new BDFS at the very end of the flow. This is to prevent the user killing the app, so I defer this operation to the very last skip and then check if we skip BDFS before and if so I ask for biometric followed by creating BDFS.
Unless you see how it can be achieved in other way then please let me know.

Reply:
understood, fair enough.